mmdet/models/losses/iou_loss.py

```diff
@@ -237,6 +237,88 @@ def ciou_loss(pred, target, eps=1e-7):
     return loss
 
 
+@weighted_loss
+def alphaiou_loss(pred, target, eps=1e-9, alpha=3, mode='iou'):
+    r"""`Implementation of paper  Alpha-IoU: A Family of Power
+    Intersection over Union Losses for Bounding Box Regression
+    <https://arxiv.org/abs/2110.13675>` _.
+
+    Code is modified from https://github.com/Jacobi93/Alpha-IoU.
+
+    Args:
+        pred (Tensor): Predicted bboxes of format (x1, y1, x2, y2),
+                    shape (n, 4).
+        target (Tensor): Corredponding gt bboxes, shape (n, 4)
+        eps (float): Eps to avoid log(0).
+        alpha (int): adaptively reweighting the loss and gradient of IoU
+        mode (str): Options are "iou"(default), "giou", "diou" and "ciou"
+    Return:
+        Tensor: Loss tensor.
+    """
+    # iou mode
+    mode = mode.lower()
+    assert mode in ('iou', 'ciou', 'giou', 'diou')
+
+    b1_x1, b1_y1 = pred[:, 0], pred[:, 1]
+    b1_x2, b1_y2 = pred[:, 2], pred[:, 3]
+    b2_x1, b2_y1 = target[:, 0], target[:, 1]
+    b2_x2, b2_y2 = target[:, 2], target[:, 3]
+
+    # overlap
+    overlap = (torch.min(b1_x2, b2_x2) - torch.max(b1_x1, b2_x1)).clamp(0) *\
+              (torch.min(b1_y2, b2_y2) - torch.max(b1_y1, b2_y1)).clamp(0)
+
+    # union
+    w1, h1 = b1_x2 - b1_x1, b1_y2 - b1_y1 + eps
+    w2, h2 = b2_x2 - b2_x1, b2_y2 - b2_y1 + eps
+    union = w1 * h1 + w2 * h2 - overlap + eps
+
+    # change conventional iou to alpha pow
+    ious = torch.pow(overlap / union + eps, alpha)
```

Comment:
We can only keep the eps at `union = w1 * h1 + w2 * h2 - overlap + eps` to avoid Nan. The other eps will not affect
 the final results.

Reply:
> We can only keep the eps at `union = w1 * h1 + w2 * h2 - overlap + eps` to avoid Nan. The other eps will not affect the final results.

`eps` here is aim to  align the accuracy with the original implementation. Obviously it's redundant, emm, okay I drop it!